source/clear-linux/tutorials/kubernetes.rst

```diff
@@ -0,0 +1,337 @@
+
+Runc and `Kata Containers`_' kata-runtime are runtimes that adhere to :abbr:`OCI
+(Open Container Initiative)` guidelines and work seamlessly with Kubernetes.
+`Kata Containers`_\* provide strong isolation for untrusted workloads or
+multi-tenant scenarios.  Runc and Kata Containers can be allocated on a
+per-pod basis so you can mix and match both on the same host to suit your needs.
+
+Prerequisites
+*************
+
+This tutorial assumes you have installed |CL| updated to the latest release
+on your host system. You can learn about the benefits for cloud orchestration
+of having an up-to-date system in the `Software Update documentation`_.
+For detailed instructions on installing |CL| on a bare metal system, follow
+the :ref:`bare metal installation tutorial<bare-metal-install>`.
+
+Before you install any new packages, update |CL| with the following command:
+
+.. code-block:: bash
+
+   sudo swupd update
+
+Install Kubernetes and CRI runtimes
+***********************************
+
+Kubernetes and a set of supported :abbr:`CRI (Container Runtime Interface)` 
+runtimes are included in the `cloud-native-basic`_ bundle. To install the 
+framework, enter the following command:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add cloud-native-basic
+
+Configure Kubernetes
+********************
+
+This tutorial uses the basic default Kubernetes configuration for simplicity.
+You must define your Kubernetes configuration according to your specific
+deployment and your security needs.
+
+#. Enable IP forwarding to avoid "preflight check" errors:
+
+   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
+   and include the following line:
+
+   .. code-block:: bash
+
+      net.ipv4.ip_forward = 1
+
+   Apply the change:
+
+   .. code-block:: bash
+
+      sudo systemctl restart systemd-sysctl
+
+#. Enable kubelet service:
+
+   .. code-block:: bash
+
+      sudo systemctl enable kubelet.service
+
+#. Disable swap using one of the following methods, either:
+
+   a) Temporarily:
+
+      .. code-block:: bash
+
+         sudo swapoff -a
+
+      .. note::
+
+         Swap will be enabled at next reboot, causing failures in
+         your cluster.
+
+   or:
+
+   b) Permanently:
+
+      Mask the swap partition:
+
+      .. code-block:: bash
+
+         sudo systemctl mask `sed -r -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps`
+         sudo swapoff -a
+
+      .. note::
+
+         On systems with limited resources, some performance degradation may
+         be observed while swap is disabled.
+         On some systems you may need to substitute the command 
+         "sudo systemctl mask `sed -r -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps`"
+         with
+         "sudo systemctl mask $(sed -n -e 's#^/dev/([0-9a-z]).#dev-\1.swap#p' /proc/swaps)"
+#. Create (or edit if it exists) the hosts file that kubernetes will read to locate
+   master's host:
```

Comment:
This is confusing to me because why can't one command work for everyone?

BTW the alternative command did not work for me either.
`
$ sudo systemctl mask $(sed -n -e 's#^/dev/([0-9a-z]).#dev-\1.swap#p' /proc/swaps)
sed: -e expression #1, char 33: invalid reference \1 on `s' command's RHS
Too few arguments.
`

Reply:
What's in your ```/proc/swaps```?

Reply:
Confirmed. :( I received the same error when running this command. "Too few arguments."

Reply:
Hold on. Both of the commands are wrong. Also, only one command is enough. There's no "on some systems..."

```sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps)```

Reply:
@mythi. Thanks for your feedback.  Please let us know as soon as you find the correct command. 
Both commands fail when I try them too.

Reply:
I have it posted above.

Reply:
```Too few arguments.``` means you already have swap disabled (from previous attempts?)

Reply:
I talked with @eadamsintel. We discussed adding a note here: 

If `Too few arguments` appears as error message, determine if `swap` is already disabled.  
Run: `cat /process/swaps`

Do you agree this note should be added?

Reply:
Or

```sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null```

Reply:
@joselamego I agree that only one command should be used. Remove the current `note` that begins with "On some systems...." Please replace existing command, after *Mask the swap partition* with @mythi 's revised command above. 

Please let me know if you think adding my note, in response to `Too few arguments`, is necessary.

Reply:
LGTM. @mvincerx: should I make the edit and push?

Reply:
Yes, please!